### ISSUE TITLE
[Feature]: Support compiler options: managedPaths and unmanagedPaths

### ISSUE THREAD

**Issue by shining-mind** [web-infra-dev/rspack#7763]
### What problem does this feature solve?

This allows for the modification of the source code of the packages inside `node_modules` using yalc or by manual means. It is particularly useful for debugging self-developed libraries.

### What does the proposed API of configuration look like?

Same as in the webpack:
- [managedPaths](https://webpack.js.org/configuration/other-options/#managedpaths)
- [unmanagedPaths](https://webpack.js.org/configuration/other-options/#unmanagedpaths)

**Comment by jerrykingxyz** [comment]
Currently rspack will watch all files, including those inside "node_modules". And this should satisfy your request.

So I understand you want rspack to ignore certain changes in node_modules by supporting managedPath and unmanagedPath?

**Comment by shining-mind** [comment]
> Currently rspack will watch all files, including those inside "node_modules". And this should satisfy your request.
> 
> So I guess you want rspack to ignore certain changes in node_modules by supporting managedPath and unmanagedPath?

Yes, watching all files inside node_modules is a heavy operation.

**Comment by jerrykingxyz** [comment]
maybe you should use https://rspack.dev/config/watch#watchoptionsignored.

**Comment by shining-mind** [comment]
> maybe you should use https://rspack.dev/config/watch#watchoptionsignored.

Snapshot managing is not associated in any way with the watching. Currently as I know rspack will rebuild when any change occurs. Initially I created the issue because I thought that rspack doesn't rebuild when dependencies inside node_modules are changed, but it does.

The option managedPaths, unmanagedPaths should be added only when the rspack will stop applying changes from the node_modules directory.

**Comment by stormslowly** [comment]
bump
